File: estimate_serial_interval.R
# Empirical Marburg serial interval from reconstructed infector-infectee pairs.
#
# Reads marburg_transmission_pairs.csv (onset dates of infector and infectee for
# documented / epidemiologically linked transmission pairs) and summarises the
# serial interval (infectee onset - infector onset) by evidence tier. A gamma
# distribution is fit by method of moments (shape = mean^2/var, scale = var/mean).
#
# Pair provenance (see the link_basis column):
#   - Tanzania 2023: transmission tree documented case-by-case in Mmbaga et al.,
#     PLoS ONE 2024 (doi:10.1371/journal.pone.0309762), mapped to onset dates in
#     MarburgTanzania2023Line.csv.
#   - Belgrade 1967, Kenya 1980: single index -> single secondary (repo INTRO flag).
#   - South Africa 1975: index -> companion and index -> nurse; star topology
#     assumed from the INTRO flags (medium confidence).
#
# CAVEATS: (1) serial interval != generation interval; (2) six of the high-tier
# pairs share one infector (the Tanzania index), so the SI *variance* is
# underestimated (shared-exposure star) and the small CV should not be read as
# the generation-interval CV; the mean is the reliable quantity.

pairs <- read.csv("marburg_transmission_pairs.csv", stringsAsFactors = FALSE)
pairs$serial_interval_days <-
  as.integer(as.Date(pairs$infectee_onset) - as.Date(pairs$infector_onset))

gamma_mom <- function(x) {
  m <- mean(x); v <- var(x)
  c(shape = m^2 / v, scale = v / m)
}
summ <- function(x) {
  g <- gamma_mom(x)
  data.frame(n_pairs = length(x), mean = round(mean(x), 2), sd = round(sd(x), 2),
             cv = round(sd(x) / mean(x), 3), median = round(median(x), 1),
             gamma_shape = round(g[["shape"]], 2), gamma_scale = round(g[["scale"]], 2),
             gamma_mean = round(g[["shape"]] * g[["scale"]], 2))
}

sets <- list("high only"    = pairs$confidence == "high",
             "high+medium"  = pairs$confidence %in% c("high", "medium"),
             "all incl low" = rep(TRUE, nrow(pairs)))

out <- do.call(rbind, Map(function(nm, idx) cbind(set = nm, summ(pairs$serial_interval_days[idx])),
                          names(sets), sets))
rownames(out) <- NULL
write.csv(out, "serial_interval_summary.csv", row.names = FALSE)
print(out)

File: estimate_serial_interval.jl
# Julia counterpart of estimate_serial_interval.R.
# Empirical Marburg serial interval from reconstructed infector-infectee pairs
# in marburg_transmission_pairs.csv; gamma fit by method of moments.
# See estimate_serial_interval.R for pair provenance and caveats
# (serial interval != generation interval; the high-tier variance is
# underestimated because most pairs share the Tanzania index case).

using CSV, DataFrames, Dates, Statistics

pairs = CSV.read("marburg_transmission_pairs.csv", DataFrame)
pairs.serial_interval_days = Dates.value.(Date.(pairs.infectee_onset) .- Date.(pairs.infector_onset))

function summ(x)
    x = Float64.(x); m = mean(x); v = var(x)
    (n_pairs = length(x), mean = round(m, digits=2), sd = round(std(x), digits=2),
     cv = round(std(x)/m, digits=3), median = round(median(x), digits=1),
     gamma_shape = round(m^2/v, digits=2), gamma_scale = round(v/m, digits=2),
     gamma_mean = round(m, digits=2))
end

sets = ["high only"    => pairs.confidence .== "high",
        "high+medium"  => in.(pairs.confidence, Ref(["high","medium"])),
        "all incl low" => trues(nrow(pairs))]

out = DataFrame()
for (nm, idx) in sets
    push!(out, merge((set = nm,), summ(pairs.serial_interval_days[idx])); cols = :union)
end
CSV.write("serial_interval_summary.csv", out)
show(out, allrows = true, allcols = true); println()
